api/jsonapi.go

```diff
@@ -1139,30 +1139,87 @@ func (i *jsonAPIHandler) GETFollowing(w http.ResponseWriter, r *http.Request) {
 }
 
 func (i *jsonAPIHandler) GETInventory(w http.ResponseWriter, r *http.Request) {
-	type inv struct {
-		Slug     string `json:"slug"`
-		Variant  int    `json:"variant"`
-		Quantity int    `json:"quantity"`
+	// Get optional peerID and slug parameters
+	var (
+		peerIDString string
+		slug         string
+	)
+
+	parts := strings.Split(r.URL.Path, "/")
+	if len(parts) > 3 {
+		peerIDString = parts[3]
 	}
-	var invList []inv
-	inventory, err := i.node.Datastore.Inventory().GetAll()
+	if len(parts) > 4 {
+		slug = parts[4]
+	}
+
+	// If we want our own inventory get it from the local database and return
+	getPersonalInventory := (peerIDString == "" || peerIDString == i.node.IPFSIdentityString())
+	if getPersonalInventory {
+		type inv struct {
+			Slug     string `json:"slug"`
+			Variant  int    `json:"variant"`
+			Quantity int    `json:"quantity"`
+		}
+		var invList []inv
+		inventory, err := i.node.Datastore.Inventory().GetAll()
+		if err != nil {
+			ErrorResponse(w, http.StatusInternalServerError, err.Error())
+			return
+		}
+		for itemSlug, m := range inventory {
+			for variant, count := range m {
+				if slug != "" && slug != itemSlug {
+					continue
+				}
+
+				i := inv{itemSlug, variant, count}
+				invList = append(invList, i)
+			}
+		}
+		ret, _ := json.MarshalIndent(invList, "", "    ")
+		if string(ret) == "null" {
+			fmt.Fprint(w, `[]`)
+			return
+		}
+		SanitizedResponse(w, string(ret))
+		return
+	}
+
+	// If we want another peer's inventory crawl IPFS with an optional cache
+	var err error
+	useCacheBool := false
+	useCacheString := r.URL.Query().Get("useCache")
```

Comment:
I believe `usecache` is lowercase in the other places where we use it.

Reply:
Good call. Fixed.